tests/src/helpers.c

```diff
@@ -244,3 +244,67 @@ void mbedtls_param_failed( const char *failure_condition,
     }
 }
 #endif /* MBEDTLS_CHECK_PARAMS */
+
+#if defined(MBEDTLS_PSA_CRYPTO_BUILTIN_KEYS)
+
+#include <psa/crypto.h>
+
+typedef struct
+{
+    psa_key_location_t location;
+    psa_drv_slot_number_t slot_number;
+} mbedtls_psa_builtin_key_description_t;
+
+static const mbedtls_psa_builtin_key_description_t builtin_keys[] = {
+    // TODO: declare some keys
+    {0, 0},
+};
+
+psa_status_t mbedtls_psa_platform_get_builtin_key(
+    psa_key_attributes_t *attributes,
+    uint8_t **p_key_buffer, size_t *key_buffer_size )
+{
+    mbedtls_svc_key_id_t svc_key_id = psa_get_key_id( attributes );
+    psa_key_id_t app_key_id = MBEDTLS_SVC_KEY_ID_GET_KEY_ID( svc_key_id );
+    size_t offset = app_key_id - MBEDTLS_PSA_KEY_ID_BUILTIN_MIN + 1;
+    if( offset >= sizeof( builtin_keys ) )
+        return( PSA_ERROR_DOES_NOT_EXIST );
+    const mbedtls_psa_builtin_key_description_t *descr = &builtin_keys[offset];
```

Comment:
This means that the builtin_keys array always needs to be as big as the largest offset, even though smaller offsets aren't mapped? This doesn't make sense to me, maybe it's better to add the ID hardcoded into the builtin_keys array too, and do a search instead of a lookup?

Reply:
Maybe. I meant this code as a demo, not as the only way to do it.